srlife/damage.py

```diff
@@ -392,30 +502,72 @@ class PIAModel(CrackShapeIndependent):
     """
 
     def calculate_element_log_reliability(
-        self, mandel_stress, temperatures, volumes, material
+        self, time, mandel_stress, temperatures, volumes, material, tot_time
     ):
         """
         Calculate the element log reliability
 
         Parameters:
+          time:           time instance variable for each stress/temperature
           mandel_stress:  element stresses in Mandel convention
           temperatures:   element temperatures
           volumes:        element volumes
           material:       material model object with required data that includes
-                          Weibull scale parameter (svals) and Weibull modulus (mvals)
+                          Weibull scale parameter (svals), Weibull modulus (mvals)
+                          and fatigue parameters (Bv,Nv)
+          tot_time:       total service time used as input to calculate reliability
         """
+
+        # Principal stresses
+        pstress = self.calculate_principal_stress(mandel_stress)
+
+        # CARES/LIFE cutoff
```

Comment:
I tried a different approach. I defined the if condition inside the calculate_principal_stress function itself so that it spits out the modified pstress.